src/server/index.ts

```diff
@@ -1,12 +1,21 @@
 import express from 'express';
 import http from 'http';
 import path from 'path';
-import { Server } from 'socket.io';
+import { Server, Socket } from 'socket.io';
 
 const app = express();
 const port = 3000;
 const server = http.createServer(app);
-const io = new Server(server);
+const io = new Server<ClientToServerEvents, ServerToClientEvents>(server);
+
+const namesToIds = new Map<string, string>();
+const clearName = (idToMatch: string) => {
+    const matchingName = [...namesToIds.entries()].find(
+        ([, id]) => id === idToMatch
+    );
+    if (!matchingName) return;
+    namesToIds.delete(matchingName[0]);
+};
```

Comment:
namesToIds could be refactored to be more testable in the future - encapsulating the value of namesToIds / the function clearName inside of exports that are tested for instance